syncMap.go

```diff
@@ -0,0 +1,43 @@
+package pongo2
+
+import (
```

Comment:
I think wht also uses a SyncMap. Do you think we should create a shared repo which we can use here as well as in wht? Will reduce duplication.

Reply:
We use are [rudder-go-kit](https://github.com/rudderlabs/rudder-go-kit) for such common functionalities. I can create a PR there